24_find_lab/findlab.txt

```diff
@@ -1,24 +1,140 @@
 1. Write command lines to find using "find":
 
   -. All files owned by yourself in /tmp folder
+  
+  all files  in /tmp/(2 users : maxim & test)
+  
+  maxim@maxim-virtual-machine:/tmp$ ls -l
+total 12
+-rw-rw-r-- 1 test  test     0 Jan  4 09:48 file1.test
+-rw-rw-r-- 1 test  test     0 Jan  4 09:48 file2.test
+drwx------ 3 root  root  4096 Jan  4 09:40 systemd-private-07794fef300641d3a64d3c789b0d3e70-colord.service-QhXkz8
+drwx------ 3 root  root  4096 Jan  4 09:40 systemd-private-07794fef300641d3a64d3c789b0d3e70-rtkit-daemon.service-cEkNto
+drwx------ 3 root  root  4096 Jan  4 09:39 systemd-private-07794fef300641d3a64d3c789b0d3e70-systemd-timesyncd.service-JiwDox
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test1.maxim
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test2.maxim
+-rw-rw-r-- 1 maxim maxim    0 Jan  4 09:42 test.maxim
+
+
+
+  maxim@maxim-virtual-machine:/tmp$ find . -exec ls -l \; | grep $(whoami) | uniq
+  
+  ** I don't understabd why it print all 4 times, and why "uniq" is not working.
+  
+  maxim@maxim-virtual-machine:/tmp$ find . -exec ls -l \; | grep $(whoami) | uniq -u
+find: ‘./systemd-private-07794fef300641d3a64d3c789b0d3e70-rtkit-daemon.service-cEkNto’: Permission denied
+-rw-rw-r-- 1 maxim maxim 11888 Jan  4 12:24 1.txt
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:02 a.txt
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:08 a.txt.bak
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:02 b.txt
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:08 b.txt.bak
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:02 c.txt
+-rw-rw-r-- 1 maxim maxim     2 Jan  4 10:08 c.txt.bak
+-rw-rw-r-- 1 maxim maxim     0 Jan  4 09:42 test1.maxim
+-rw-rw-r-- 1 maxim maxim     0 Jan  4 09:42 test2.maxim
+-rwxrwxrwx 1 maxim maxim     0 Jan  4 09:42 test.maxim
+
+It prints all X4 times
+  
+  
 
   -. All files and directories not owned by you in /tmp folder
+  
+  user root,test
```

Comment:
there's a cool neat feature to ```find``` that allows you to invert the next flag it's called ```-not``` so this command:
``` find ./ -not -user maxim``` 
will print all the files that are not owned by you.